docs/upgrade.html

```diff
@@ -224,17 +224,19 @@ <h5><a id="upgrade_400_notable" href="#upgrade_400_notable">Notable changes in 4
         </li>
         <li>Other changes:
             <ul>
+                <li>
+                    The minimum Java version required by clients and streams applications has been increased from Java 8 to Java 11
+                    while brokers, connect and tools now require Java 17.
+                    See <a href="https://cwiki.apache.org/confluence/pages/viewpage.action?pageId=181308223">KIP-750</a> and
+                    <a href="https://cwiki.apache.org/confluence/pages/viewpage.action?pageId=284789510">KIP-1013</a> for more details.
```

Comment:
I moved this higher up because it's more impactful than the other notes in this section.